test/test_view_isodoses_tab.py

```
@@ -0,0 +1,107 @@
+import os
+import pytest
+from pathlib import Path
+
+from src.Controller.GUIController import MainWindow
+from src.Model.PatientDictContainer import PatientDictContainer
+from src.View.mainpage.IsodoseTab import isodose_percentages
+from src.Model import ImageLoading
+
+from pydicom import dcmread
+from pydicom.errors import InvalidDicomError
+
+
+def find_DICOM_files(file_path):
+    """
+    Function to find DICOM files in a given folder.
+    :param file_path: File path of folder to search.
+    :return: List of file paths of DICOM files in given folder.
+    """
+
+    dicom_files = []
+
+    # Walk through directory
+    for root, dirs, files in os.walk(file_path, topdown=True):
+        for name in files:
+            # Attempt to open file as a DICOM file
+            try:
+                dcmread(os.path.join(root, name))
+            except (InvalidDicomError, FileNotFoundError):
+                pass
```

Given a known test input (set of files), wouldn't you know which files should fail?
It's arguable whether one should rethrow the exception if a failure is really not expected, but it at least implies something is wrong with the data.

---

It was written this way because the current test data set includes some non-DICOM files. I also personally believe that finding the test DICOM files programmatically is better than hard-coding their locations, as it future-proofs the test in the event that the test data changes in the future. The test dataset on my local machine is also larger than what is in this repository.

---

OK. Your test currently specifies a particular data set but you expect alternative data in the future.
My thinking was to identify known exceptions as a secondary filter after a failed read rather than identifying known good data explicitly.
Just wanted a documented discussion of the choices being made.  No wrong answer...

---

That's a good idea, I'll look into implementing that for any future tests I write that involve test data. Thank you for the feedback.